jobs/e2e-test-cycloud.groovy

```diff
@@ -152,30 +154,74 @@ def runAllTestClients() {
    for (i = 0; i < NUM_WORKER_MACHINES; i++) {
       def workerId = i;  // avoid scoping problems
       jobs["e2e-test-${workerId}"] = {
-         swallowExceptions({
+         stage("e2e-worker-${workerId}") {
             onWorker(WORKER_TYPE, '2h') {
-                _setupWebapp();
-                runE2ETests(workerId);
+               _setupWebapp()
+               runE2ETests(workerId)
+               dir("${REPORT_DIR}") {
+                  stash includes: "e2e-test-results.json", name: "worker-${workerId}-reports"
+               }
             }
-         }, onException);
-      };
-   }
+         }
+      }
+   };
    parallel(jobs);
 }
 
 def runE2ETests(workerId) {
    echo("Starting e2e tests for worker ${workerId}");
 
-   // Determine which environment we're running against, so we can provide a tag
-   // in the LambdaTest build.
+   // Define which environment we're running against, and setting up junit report
    def e2eEnv = E2E_URL == "https://www.khanacademy.org" ? "prod" : "preprod";
 
-   // NOTE: We hard-code the values here as it is an experiment that will be
-   // removed soon.
-   def runE2ETestsArgs = ["env", "CYPRESS_PROJECT_ID=2c1iwj", "CYPRESS_RECORD_KEY=4c530cf3-79e5-44b5-aedb-f6f017f38cb5", "./dev/cypress/e2e/tools/start-cypress-cloud-run.ts"];
+   def runE2ETestsArgs = [
+           "./dev/cypress/e2e/tools/start-cy-cloud-run.ts",
+           "--url=${E2E_URL}",
+           "--name=${BUILD_NAME}",
+   ];
 
    dir('webapp/services/static') {
-      exec(runE2ETestsArgs);
+      exec(runE2ETestsArgs);}
+   }
+}
+
+def unstashReports() {
+   def jsonFolders = []
+   dir("${REPORT_DIR}") {
+      for (i = 0; i < NUM_WORKER_MACHINES; i++) {
+         sh "mkdir -p ./${i}"
```

Comment:
```suggestion
         exec(["mkdir", "-p", "${i}"]);
```
But I think you need to do an `rmdir` first.  Otherwise, the results from a previous run will still be sticking around here.

Reply:
Yes, I believe something in this job somewhere needs to clean up old results files so that we're only reporting on this specific run.